### PR TITLE
Disable warning for concat. This fixes #6715

### PR DESCRIPTION
**Pull request by 3cham**
Hello, thanks for working on this great framework. This is my first try to contribute to it.

- [x] Tests added / passed
- [x] Passes `black dask` / `flake8 dask`

### CODE REVIEW

File: dask/dataframe/tests/test_multi.py [hunk 1]
```diff
@@ -1406,6 +1406,10 @@ def test_concat_unknown_divisions():
     with pytest.raises(ValueError):
         dd.concat([aa, cc], axis=1)
 
+    with warnings.catch_warnings(record=True) as w:
+        dd.concat([aa, bb], axis=1, ignore_unknown_divisions=True)
+        assert len(w) == 0
```
Comment (jcrist) [inline dask/dataframe/tests/test_multi.py:1411]: For uniformity, this should use [`pytest.warns`](https://docs.pytest.org/en/latest/warnings.html#recwarn) instead:

```suggestion
    with pytest.warns(None) as record:
        dd.concat([aa, bb], axis=1, ignore_unknown_divisions=True)
    assert len(record) == 0
```

File: dask/dataframe/multi.py [hunk 2]
```diff
@@ -1013,6 +1019,9 @@ def concat(dfs, axis=0, join="outer", interleave_partitions=False):
     interleave_partitions : bool, default False
         Whether to concatenate DataFrames ignoring its order. If True, every
         divisions are concatenated each by each.
+    ignore_unknown_divisions: boolean, default False
+        Whether to warn when concatenating dask series/dataframes
+        with unknown divisions. If True the warning won't be emitted.
```
Comment (jcrist) [inline dask/dataframe/multi.py:1024]: This option is the opposite, whether to *not* warn:

```suggestion
        By default a warning is raised if any input has unknown divisions.
        Set to True to disable this warning.
```

**Comment by 3cham** [comment]
@jcrist thanks for your suggestions, I've added in the following commit.

**Comment by TomAugspurger** [comment]
CI failure is unrelated. Merging.

Thanks @3cham!

**Pull request merged**